packages/autoloader/src/CustomAutoloaderPlugin.php

```diff
@@ -0,0 +1,88 @@
+<?php
+/**
+ * Custom Autoloader Composer Plugin, hooks into composer events to generate the custom autoloader.
+ *
+ * @package Automattic\Jetpack\Autoloader
+ */
+
+// phpcs:disable PHPCompatibility.Keywords.NewKeywords.t_useFound
+// phpcs:disable PHPCompatibility.LanguageConstructs.NewLanguageConstructs.t_ns_separatorFound
+// phpcs:disable PHPCompatibility.Keywords.NewKeywords.t_namespaceFound
+// phpcs:disable WordPress.Files.FileName.NotHyphenatedLowercase
+// phpcs:disable WordPress.Files.FileName.InvalidClassFileName
+// phpcs:disable WordPress.NamingConventions.ValidVariableName.VariableNotSnakeCase
+
+namespace Automattic\Jetpack\Autoloader;
+
+use Composer\Composer;
+use Composer\IO\IOInterface;
+use Composer\Script\Event;
+use Composer\Script\ScriptEvents;
+use Composer\Plugin\PluginInterface;
+use Composer\EventDispatcher\EventSubscriberInterface;
+
+/**
+ * Class CustomAutoloaderPlugin.
+ *
+ * @package Automattic\Jetpack\Autoloader
+ */
+class CustomAutoloaderPlugin implements PluginInterface, EventSubscriberInterface {
+
+	/**
+	 * IO object.
+	 *
+	 * @var IOInterface IO object.
+	 */
+	private $io;
```

Comment:
Do we need these?

Reply:
I think so... we need to pass it into the AutoGenerator instance.